src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/læremidler/beregning/LæremidlerBeregningService.kt

```diff
@@ -1,49 +1,132 @@
 
-import no.nav.tilleggsstonader.kontrakter.felles.splitPerMåned
-import no.nav.tilleggsstonader.sak.vedtak.læremidler.domain.BeregningPeriode
+import no.nav.tilleggsstonader.kontrakter.felles.mergeSammenhengende
+import no.nav.tilleggsstonader.kontrakter.felles.overlapper
+import no.nav.tilleggsstonader.sak.felles.domain.BehandlingId
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.feilHvis
+import no.nav.tilleggsstonader.sak.vedtak.barnetilsyn.domain.Stønadsperiode
+import no.nav.tilleggsstonader.sak.vedtak.barnetilsyn.domain.tilSortertGrunnlagStønadsperiode
+import no.nav.tilleggsstonader.sak.vedtak.læremidler.beregning.LæremidlerBeregningUtil.delTilUtbetalingsPerioder
 import no.nav.tilleggsstonader.sak.vedtak.læremidler.domain.Beregningsgrunnlag
 import no.nav.tilleggsstonader.sak.vedtak.læremidler.domain.BeregningsresultatForMåned
 import no.nav.tilleggsstonader.sak.vedtak.læremidler.domain.BeregningsresultatLæremidler
+import no.nav.tilleggsstonader.sak.vedtak.læremidler.domain.Vedtaksperiode
+import no.nav.tilleggsstonader.sak.vilkår.stønadsperiode.domain.StønadsperiodeRepository
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.MålgruppeType
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.ResultatVilkårperiode
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.VilkårperiodeRepository
 import org.springframework.stereotype.Service
 import java.math.BigDecimal
 import java.math.RoundingMode
-import java.time.YearMonth
 
 private val PROSENT_50 = BigDecimal(0.5)
 private val PROSENTGRENSE_HALV_SATS = 50
 
 @Service
-class LæremidlerBeregningService {
+class LæremidlerBeregningService(
+    private val vilkårperiodeRepository: VilkårperiodeRepository,
+    private val stønadsperiodeRepository: StønadsperiodeRepository,
+) {
     /**
      * Beregning av læremidler er foreløpig kun basert på antakelser.
      * Nå beregnes det med hele måneder, splittet i månedsskifte, men dette er ikke avklart som korrekt.
      * Det er ikke tatt hensyn til begrensninger på semester og maks antall måneder i et år som skal dekkes.
      */
 
-    fun beregn(periode: BeregningPeriode): BeregningsresultatLæremidler {
-        val beregningsgrunnlagPerMåned = periode.splitPerMåned { måned, periode ->
-            val beregningsGrunnlag = lagBeregningsGrunnlag(periode, måned)
-            BeregningsresultatForMåned(
-                beløp = finnBeløpForStudieprosent(beregningsGrunnlag.sats, beregningsGrunnlag.studieprosent),
-                grunnlag = lagBeregningsGrunnlag(periode, måned),
+    // ANTAR: En aktitivet hele vedtaksperioden
+    // ANTAR: En stønadsperiode per vedtaksperiode
+    // ANTAR: Sats ikke endrer seg i perioden
+    fun beregn(vedtaksperioder: List<Vedtaksperiode>, behandlingId: BehandlingId): BeregningsresultatLæremidler {
+        val stønadsperioder =
+            stønadsperiodeRepository.findAllByBehandlingId(behandlingId).tilSortertGrunnlagStønadsperiode()
```

Comment:
Kan vurdere å trekke ut henting av stønadsperiode til egen metode og slå sammen de periodene i den metoden. Så trenger man ikke gjøre det for **hver** vedtaksperiode i `finnRelevantStønadsperiode` og gjør `finnRelevantStønadsperiode` litt enklere

Reply:
Det kan funke bra 👍  0a579e730d1a5f4198eccd555d7f2a50adc3277d